packages/core/src/nodes.rs

```diff
@@ -48,7 +48,10 @@ pub struct VNode<'a> {
     /// The key given to the root of this template.
     ///
     /// In fragments, this is the key of the first child. In other cases, it is the key of the root.
-    pub key: Option<&'a str>,
+    ////////////////////////////////////////////////////////////////////////////
+    //////////////////// Change the key type in the node ///////////////////////
+    ////////////////////////////////////////////////////////////////////////////
+    pub key: Option<&'a KeyValue<'a>>,
```

Comment:
This could be changed to not borrow KeyValue. The old version borrowed the string because it does not have know size, but KeyValue is Copy and Sized so it is no longer needed